tests/unit/utils/Utils.h

```diff
@@ -55,4 +58,44 @@ SGStringList<T> generateRandomStringData(
 	return strings;
 }
 
+struct NeuralLayerTestUtil
```

Comment:
this code is so specific, I suggest we move it into a helper file within `neuralnets`